ksql-cli/src/main/java/io/confluent/ksql/util/CmdLineUtil.java

```diff
@@ -0,0 +1,199 @@
+/*
+ * Copyright 2019 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License; you may not use this file
+ * except in compliance with the License.  You may obtain a copy of the License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.util;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableList.Builder;
+import java.util.List;
+
+public final class CmdLineUtil {
+
+  private CmdLineUtil() {
+  }
+
+  /**
+   * Split the supplied line by unquoted whitespace.
+   *
+   * <p>White space within an open single quote will be ignored
+   *
+   * <p>A quote can be escaped using a second single quote, e.g. {@code "'escaped '' quote'"}.
+   *
+   * <ul>
+   * <li>{@code "t0 t1"} becomes {@code ["t0", "t1"]}</li>
+   * <li>{@code "'quoted string'"} becomes {@code ["quoted string"]}</li>
+   * <li>{@code "'quoted'connected'quoted'"} becomes {@code ["quotedconnectedquoted"]}</li>
+   * <li>{@code "'escaped '' quote'"} becomes {@code ["escaped ' quote"]}</li>
+   * </ul>
+   *
+   * @return the list of parts.
+   */
+  public static List<String> splitByUnquotedWhitespace(final String string) {
+    return new WhitespaceParser().parse(string);
+  }
+
+  /**
+   * Remove any matched, i.e. closed, single quotes from the supplied {@code string}.
+   *
+   * <p>Works in a similar way to a unix command line would.
+   *
+   * <p>A quote can be escaped using a second single quote, e.g. {@code "'escaped '' quote'"}.
+   *
+   * <p>e.g.
+   * <ul>
+   * <li>{@code "unquoted"} becomes {@code "unquoted"}</li>
+   * <li>{@code "'some quoted string'"} becomes {@code "some quoted string"}</li>
+   * <li>{@code "'quoted'connected'quoted'"} becomes {@code "quotedconnectedquoted"}</li>
+   * <li>{@code "'escaped '' quote'"} becomes {@code "escaped ' quote"}</li>
+   * </ul>
+   *
+   * @return the input with matched quotes removed.
+   */
+  public static String removeMatchedSingleQuotes(final String string) {
+    return new QuoteRemover().parse(string);
+  }
+
+  /**
+   * Removes any trailing semi colon(s).
+   *
+   * @param s the string to trim.
+   * @return the trimmed string
+   */
+  public static String trimTrailingSemiColon(final String s) {
```

Comment:
this is equivalent to `StringUtils#stripEnd(s, ";")` if I understand correctly

Reply:
nice. Wasn't aware of this method. Updated.